packages/desktop-client/src/components/common/ToggleMenu.tsx

```diff
@@ -0,0 +1,203 @@
+import { type ReactNode, useRef, useState, useEffect } from 'react';
```

Comment:
Is it it possible to merge this component with the existing `Menu` component? We could create a specialized `TogglableMenuItem` which extends from `MenuItem` which when rendered, will add a toggle component on the rendered menu item.